diff -dubr source/doc/node.1 rename/doc/node.1
--- source/doc/node.1	2012-10-26 00:49:56.000000000 +0400
+++ rename/doc/node.1	2012-11-01 21:57:48.271896703 +0400
@@ -2,12 +2,12 @@
 
 
 .SH "NAME"
-node \- Server-side JavaScript
+nodejs \- Server-side JavaScript
 
 .SH SYNOPSIS
 
 
-.B node
+.B nodejs
 [
 .B \-v
 ]
diff -dubr source/tools/wafadmin/Tools/node_addon.py rename/tools/wafadmin/Tools/node_addon.py
--- source/tools/wafadmin/Tools/node_addon.py	2012-10-26 00:49:32.000000000 +0400
+++ rename/tools/wafadmin/Tools/node_addon.py	2012-11-01 21:59:05.441880615 +0400
@@ -26,10 +26,10 @@
   conf.env['PREFIX_NODE'] = get_prefix()
   prefix = conf.env['PREFIX_NODE']
   lib = join(prefix, 'lib')
-  nodebin = join(prefix, 'bin', 'node')
+  nodebin = join(prefix, 'bin', 'nodejs')
 
   conf.env['LIBPATH_NODE'] = lib
-  conf.env['CPPPATH_NODE'] = join(prefix, 'include', 'node')
+  conf.env['CPPPATH_NODE'] = join(prefix, 'include', 'nodejs')
 
   conf.env.append_value('CPPFLAGS_NODE', '-D_GNU_SOURCE')
 
