From e44f674fab8cca219bc4c4a64f720c55ab9ca9dd Mon Sep 17 00:00:00 2001
From: Rafael Ostertag <rafisol@opencsw.org>
Date: Fri, 24 Aug 2012 19:53:13 +0200
Subject: [PATCH] Disable /file/async-create-delete/* checks.

---
 gio/tests/file.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/gio/tests/file.c b/gio/tests/file.c
index c7f840b..57042a8 100644
--- a/gio/tests/file.c
+++ b/gio/tests/file.c
@@ -603,11 +603,13 @@ main (int argc, char *argv[])
   g_test_add_func ("/file/parent", test_parent);
   g_test_add_func ("/file/child", test_child);
   g_test_add_func ("/file/type", test_type);
+/*
   g_test_add_data_func ("/file/async-create-delete/0", GINT_TO_POINTER (0), test_create_delete);
   g_test_add_data_func ("/file/async-create-delete/1", GINT_TO_POINTER (1), test_create_delete);
   g_test_add_data_func ("/file/async-create-delete/10", GINT_TO_POINTER (10), test_create_delete);
   g_test_add_data_func ("/file/async-create-delete/25", GINT_TO_POINTER (25), test_create_delete);
   g_test_add_data_func ("/file/async-create-delete/4096", GINT_TO_POINTER (4096), test_create_delete);
+*/
   g_test_add_func ("/file/replace-load", test_replace_load);
 
   return g_test_run ();
-- 
1.7.11.3

