From 53f22e4878dab62282978d1cf631a8f74a22f73b Mon Sep 17 00:00:00 2001
From: Rafael Ostertag <raos@opencsw.org>
Date: Sat, 25 Aug 2012 19:00:54 +0200
Subject: [PATCH] Fix locale for collate test

---
 glib/tests/collate.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/glib/tests/collate.c b/glib/tests/collate.c
index 29a62ea..737bfe8 100644
--- a/glib/tests/collate.c
+++ b/glib/tests/collate.c
@@ -187,9 +187,9 @@ main (int argc, char *argv[])
 
   g_test_init (&argc, &argv, NULL);
 
-  g_setenv ("LC_ALL", "en_US", TRUE);
+  g_setenv ("LC_ALL", "en_US.UTF-8", TRUE);
   locale = setlocale (LC_ALL, "");
-  if (locale == NULL || strcmp (locale, "en_US") != 0)
+  if (locale == NULL || strcmp (locale, "en_US.UTF-8") != 0)
     {
       g_test_message ("No suitable locale, skipping test");
       return 0;
-- 
1.7.11.3

