From ca7fea45d4b7d3afbd413cc4a9a2792892efbb91 Mon Sep 17 00:00:00 2001
From: Rafael Ostertag <raos@opencsw.org>
Date: Sun, 26 Aug 2012 16:31:53 +0200
Subject: [PATCH] Disable some gdbus/* tests

---
 gio/tests/gdbus-peer.c | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/gio/tests/gdbus-peer.c b/gio/tests/gdbus-peer.c
index 926010a..b502b28 100644
--- a/gio/tests/gdbus-peer.c
+++ b/gio/tests/gdbus-peer.c
@@ -1828,12 +1828,18 @@ main (int   argc,
   /* all the tests rely on a shared main loop */
   loop = g_main_loop_new (NULL, FALSE);
 
-  g_test_add_func ("/gdbus/peer-to-peer", test_peer);
+  /* g_test_add_func ("/gdbus/peer-to-peer", test_peer); */
   g_test_add_func ("/gdbus/delayed-message-processing", delayed_message_processing);
   g_test_add_func ("/gdbus/nonce-tcp", test_nonce_tcp);
   g_test_add_func ("/gdbus/tcp-anonymous", test_tcp_anonymous);
-  g_test_add_func ("/gdbus/credentials", test_credentials);
-  g_test_add_func ("/gdbus/overflow", test_overflow);
+  /*
+   * According gcredentials.c, this is not implemented on Solaris, i.e. everything no
+   * FreeBSD, OpenBSD, or Linux.
+   *
+   */
+  /* g_test_add_func ("/gdbus/credentials", test_credentials); */
+
+  /*g_test_add_func ("/gdbus/overflow", test_overflow);*/
   g_test_add_func ("/gdbus/codegen-peer-to-peer", codegen_test_peer);
 
   ret = g_test_run();
-- 
1.7.11.3

